elide-core/src/main/java/com/yahoo/elide/security/Check.java

```diff
@@ -15,9 +17,35 @@
 public interface Check<T> {
 
     /**
-     * Determines whether the user can access the resource.
-     * @param record the record
+     * Determines whether the user can access the resource. This method is called specifically directly before the
+     * commit. That is, if full/reliable information about an object (based on this request) is required, this is the
+     * method which should be implemented.
+     *
+     * <b>NOTE:</b> The object on this method is only complete for Create, Update, and Share permissions. If used
+     *              in a Read or Delete permission, the final object values are not guaranteed since
+     *              they will not be deferred until commit time. That is, this is treated as an additional
+     *              <i>operation check</i> in the cases of Read and Delete permissions. Moreover, it is joined in
+     *              conjunction with the result from the proper operation check in such a scenario.
+     *
+     * @param object Fully modified object
+     * @param requestScope Request scope object
+     * @param changeSpec Summary of modifications
+     * @return true if allowed
+     */
+    default boolean ok(T object, RequestScope requestScope, Optional<ChangeSpec> changeSpec) {
```

Comment:
I'm wondering if we should have different ok methods (rather than optional here) for cases where we need to check field level permissions.  The changeSpec only really makes sense for fields and relationships.